src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -294,16 +294,15 @@ async def validate_rsync_task(self, data, schema):
 
                             # If for whatever reason the dir does not exist, let's create it
                             # An example of this is when we run rsync tests we nuke the directory
-                            if not await self.middleware.run_in_thread(ssh_dir_path.exists):
-                                # We still do a safe mkdir to avoid any race conditions
-                                await self.middleware.run_in_thread(ssh_dir_path.mkdir, 0o700, False, True)
-                                await self.middleware.run_in_thread(
-                                    os.chown, ssh_dir_path.absolute(), user['pw_uid'], user['pw_gid']
-                                )
+                            def handle_ssh_dir():
+                                with contextlib.suppress(FileExistsError):
+                                    ssh_dir_path.mkdir(0o700)
+                                    os.chown(ssh_dir_path.absolute(), user['pw_uid'], user['pw_gid'])
```

Comment:
This os.chown call will have to be moved outside of the scope of the `with contextlib` since if it crashes with `FileExistsError`, the `os.chown` won't be made

Reply:
This is intended, why chown directory if it already exists?

Reply:
Because it could be owned by a different uid/gid